src/ZServer/HTTPServer.py

```diff
@@ -235,7 +235,7 @@ def get_environment(self, request,
             value=value.strip()
             if h2ehas(key) and value:
                 env[h2eget(key)]=value
-            else:
+            elif "_" not in key:
```

Comment:
Ugh, this module has no existing test coverage!

Can you add a comment explaining the rationale, e.g.:

```suggestion
            elif "_" not in key:  # Headers with underscores, might spoof real ones.
```

Reply:
I changed it a little not to exceed 80 chars.